pycontrails/models/humidity_scaling.py

```diff
@@ -553,10 +558,239 @@ def scale(
+        ERA5-derived RHi values for the given ensemble member.
+    member : int
+        The ERA5 ensemble member to use. Must be in the range [0, 10).
+
+    Returns
+    -------
+    npt.NDArray[np.float64]
+        The IAGOS quantiles corresponding to the ERA5-derived RHi values.
+    """
+
+    era5_quantiles = _load_era5_ensemble_quantiles()  # shape (801, 10)
+    era5_quantiles = era5_quantiles[:, member]  # shape (801,)
+    iagos_quantiles = _load_iagos_quantiles()  # shape (801,)
+
+    return np.interp(era5_rhi, era5_quantiles, iagos_quantiles)
+
+
+def recalibrate_rhi(
+    era5_rhi_all_members: npt.NDArray[np.float_], member: int
+) -> npt.NDArray[np.float_]:
+    """Recalibrate ERA5-derived RHi values to IAGOS quantiles.
+
+    This recalibration requires values for **all** ERA5 ensemble members.
+
+    Parameters
+    ----------
+    era5_rhi_all_members : npt.NDArray[np.float_]
+        ERA5-derived RHi values for all ensemble members. This array should have shape (n, 10).
+    member : int
+        The ERA5 ensemble member to use. Must be in the range [0, 10).
+
+    Returns
+    -------
+    npt.NDArray[np.float_]
+        The recalibrated RHi values. This is an array of shape (n,).
+    """
+
+    n_members = 10
+    assert era5_rhi_all_members.shape[1] == n_members
+
+    # Perform histogram matching on the given ensemble member
+    recalibrated_rhi = quantile_rhi_map(era5_rhi_all_members[:, member], member)
+
+    # Perform histogram matching on all other ensemble members
+    # Add up the results into a single 'ensemble_mean_rhi' array
+    ensemble_mean_rhi: npt.NDArray[np.float_] = 0.0  # type: ignore[assignment]
+    for r in range(n_members):
+        if r == member:
+            ensemble_mean_rhi += recalibrated_rhi
+        else:
+            ensemble_mean_rhi += quantile_rhi_map(era5_rhi_all_members[:, r], r)
+
+    # Divide by the number of ensemble members to get the mean
+    ensemble_mean_rhi /= n_members
+
+    eckel_a = -0.005213832567192828
+    eckel_c = 2.7859172756970354
+
+    out = (ensemble_mean_rhi - eckel_a) + eckel_c * (recalibrated_rhi - ensemble_mean_rhi)
+    return out.astype(era5_rhi_all_members.dtype)
+
+
+@dataclasses.dataclass
+class HistogramMatchingWithEckelParams(ModelParams):
+    """Parameters for :class:`HistogramMatchingWithEckel`."""
+
+    #: A length-10 list of ERA5 ensemble members.
+    #: Each element is a :class:`MetDataArray` holding specific humidity
+    #: values for a single ensemble member. If None, a ValueError will be
+    #: raised at model evaluation time. The order of the list must be
+    #: consistent with the order of the ERA5 ensemble members.
+    ensemble_specific_humidity: list[MetDataArray] | None = None
+
+    #: The specific member used. Must be in the range [0, 10). If None,
+    #: a ValueError will be raised at model evaluation time.
+    member: int | None = None
+
+
+class HistogramMatchingWithEckel(HumidityScaling):
```

Comment:
I would prefix this with `_` and put a note in the docstring that its experimental

```
.. warning ::

```

Reply:
I included the warning, but not the prefix underscore!

4febb864eb18a0c6ec940d7535d2b92ca8f2812d